scripts/gen-api-gtkdoc.py

```diff
@@ -0,0 +1,415 @@
+            if n.tag == "computeroutput":
+                s += self.at.cb("c", self.__process_element(n))
+            if n.tag == "itemizedlist":
+                s += "\n" + self.__process_element(n)
+            if n.tag == "listitem":
+                s += " - " + self.__process_element(n)
+            if n.tag == "para":
+                s += self.__process_element(n) + "\n"
+            if n.tag == "ref":
+                s += n.text if n.text else ""
+            if n.tag == "simplesect":
+                ss = self.at.cb(n.get("kind"), self.__process_element(n))
+                s += ss + "\n" if ss else ""
+            if n.tag == "programlisting":
+                s += self.get_program_listing(n)
+            if n.tag == "xrefsect":
+                s += self.__process_element(n)
+            if n.tag == "xreftitle":
+                s += self.__process_element(n) + ": "
+            if n.tag == "xrefdescription":
+                s += self.__process_element(n)
+            if n.tag == "ulink":
+                s += self.__process_element(n)
+            if n.tag == "linebreak":
+                s += "\n"
+            if n.tag == "ndash":
+                s += "--"
+                # workaround for doxygen bug #646002
+            if n.tag == "htmlonly":
+                s += ""
+            if n.tail:
+                s += n.tail
+            if n.tag.startswith("param"):
+                pass # parameters are handled separately in DoxyFunction::from_memberdef()
+        return s
+
+class DoxyMember(object):
+    def __init__(self, name, brief, extra = ""):
+        self.name       = name
+        self.brief      = brief
+        self.extra      = extra
+
+
+class DoxyElement(object):
+
+    def __init__(self, name, definition, **kwargs):
+        self.name       = name
+        self.definition = definition
+        self.brief      = kwargs.get('brief', "")
+        self.detail     = kwargs.get('detail', "")
+        self.members    = kwargs.get('members', [])
+        self.since      = kwargs.get('since', "")
+        self.extra      = kwargs.get('extra', "")
+        self.retval     = kwargs.get('retval', None)
+
+    def is_documented(self):
+        if (normalize_text(self.brief)) != "":
+            return True
+        return False
+
+    def add_brief(self, xml):
+        proc = DoxygenProcess()
+        self.brief = proc.process_element(xml)
+        self.extra += proc.get_extra()
+
+    def add_detail(self, xml):
+        proc = DoxygenProcess()
+        self.detail = proc.process_element(xml)
+        self.extra += proc.get_extra()
+        self.since = proc.get_since()
+
+    def add_member(self, xml):
+        name = xml.find("name").text
+        proc = DoxygenProcess()
+        brief = proc.process_element(xml.find("briefdescription"))
+        # optional doxygen command output appears within <detaileddescription />
+        proc.process_element(xml.find("detaileddescription"))
```

Comment:
Mmm'kay, just mentioned that because I tried to add the @type before the brief and it was missing from the output.

But it may be bad some annotations might just disappear, which would create weird bugs in the generated output.